### PR TITLE
refactor(localization): Consolidate mappings to one config object

### PR DESCRIPTION
**Pull request by jaredvu**
Clean up localization code and use a single config object.
Allow deployers to set configure available locales from those supported.

---

## Views

- `<LanguageSelector>`, `<Settings>`
  - update to use new config 

## Constants/Types

- `constants/localization`
  - Add localization config

## Environment

- `.env.example`
  - add `VITE_APP_RESTRICTED_LOCALES`. No spaces
  - example:
    - `VITE_APP_RESTRICTED_LOCALES=en,zh-CN,fr,de`

## Functions/ Lib

- `consistentAssetSizing`
  - update to use new config 

- `lib/objectHelpers`
  - add typed object.fromEntries 

## Hooks

- `useTradingView`, `useTradingViewLaunchable`, `useLocaleSeparators`
  - update to use new config 

## State

- `localizationMiddleware`
  - update to use new config 

---

### CODE REVIEW

File: src/lib/consistentAssetSize.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import { SUPPORTED_LOCALE_STRING_LABELS, SupportedLocales } from '@/constants/localization';
+import { SUPPORTED_LOCALES, SupportedLocales } from '@/constants/localization';
 
 import { formatNumberOutput, OutputType } from '@/components/Output';
 
 // for each locale, an array of the correct compact number suffix to use for 10^{index}
 // e.g. for "en" we have ['', '', '', 'k', 'k', 'k', 'm', 'm', 'm', 'b', 'b', 'b', 't', 't', 't']
 const supportedLocaleToCompactSuffixByPowerOfTen = mapValues(
-  SUPPORTED_LOCALE_STRING_LABELS,
+  Object.fromEntries(SUPPORTED_LOCALES.map(({ locale, label }) => [locale, label])),
```
Comment (jaredvu) [inline src/lib/consistentAssetSize.ts:10]: @tyleroooo  kept this logic the same by passing `label` to `Intl.NumberFormat`. let me know if its better to pass `browserLanguage` instead

File: src/hooks/tradingView/useTradingView.ts [hunk 2]
```diff
@@ -146,6 +146,8 @@ export const useTradingView = ({
     if (marketId && tickSizeDecimals !== undefined && !tvWidget) {
       const widgetOptions = getWidgetOptions();
       const widgetOverrides = getWidgetOverrides({ appTheme, appColorMode });
+      const languageCode =
+        SUPPORTED_LOCALES.find(({ locale }) => locale === selectedLocale)?.baseTag ?? 'en';
```
Comment (tyleroooo) [inline src/hooks/tradingView/useTradingView.ts:150]: I used to entertain myself by trying to make maximally functional and type safe version of this kind of statement. Like imagine a `SUPPORTED_LOCALES.find(property('locale').eq(selecedLocale))?.baseTag`

File: src/lib/consistentAssetSize.ts (hunk 1 above)
Comment (tyleroooo) [reply src/lib/consistentAssetSize.ts:10]: I don't think that's the same. `mapValues` here is passes `(value, key)` and we ignore `value` (`name`) and use `key` (`lang`) which is a `SupportedLocale` (now `SUPPORTED_LOCALES.locale`). Instead let's replace `mapValues` with `SUPPORTED_LOCALES.map(({ locale: lang }) => `

File: src/hooks/tradingView/useTradingViewLaunchable.ts [hunk 3]
```diff
@@ -52,13 +52,14 @@ export const useTradingViewLaunchable = ({
     if (marketId && !isDataLoading && !tvWidget) {
       const widgetOptions = getWidgetOptions(true);
       const widgetOverrides = getWidgetOverrides({ appTheme, appColorMode });
+      const languageCode = SUPPORTED_LOCALE_MAP[selectedLocale].baseTag;
```
Comment (tyleroooo) [inline src/hooks/tradingView/useTradingViewLaunchable.ts:55]: will these crash if the user has a (now blacklsited) locale saved in their local state?

File: src/lib/consistentAssetSize.ts [hunk 4]
```diff
@@ -1,25 +1,26 @@
...
+import { SUPPORTED_LOCALES, SupportedLocales } from '@/constants/localization';
 
 import { formatNumberOutput, OutputType } from '@/components/Output';
 
 // for each locale, an array of the correct compact number suffix to use for 10^{index}
 // e.g. for "en" we have ['', '', '', 'k', 'k', 'k', 'm', 'm', 'm', 'b', 'b', 'b', 't', 't', 't']
-const supportedLocaleToCompactSuffixByPowerOfTen = mapValues(
-  SUPPORTED_LOCALE_STRING_LABELS,
-  (name, lang) =>
+const supportedLocaleToCompactSuffixByPowerOfTen = Object.fromEntries(
```
Comment (tyleroooo) [inline src/lib/consistentAssetSize.ts:9]: oh you're totally right, my suggestion was wrong and this is correct. Nice

**Review by tyleroooo — APPROVED** [review]

**Pull request merged**